engine/src/conversion/analysis/fun/mod.rs

```diff
@@ -952,9 +952,19 @@ impl<'a> FnAnalyzer<'a> {
                         Virtualness::PureVirtual => MethodKind::PureVirtual(receiver_mutability),
                     }
                 };
-                // Disambiguate overloads.
-                let rust_name = predetermined_rust_name
-                    .unwrap_or_else(|| self.get_overload_name(ns, type_ident, rust_name));
+
+                let rust_name = if matches!(
+                    fun.special_member,
+                    Some(SpecialMemberKind::GeneratedAccessor)
+                ) {
```

Comment:
Could you explain why this `if` clause is necessary? What does the Rust name end up being otherwise?

Reply:
Without the `if`, the rust name will still have the struct name prefix (we can see this in the current test for now since I reverted this while we think of a better strategy for determining which accessors were generated: https://github.com/google/autocxx/pull/1133/files#diff-7d1a694c0fef109924a3e6d7c3462f13a4727cfa3b2a2f86620734ca3e1800fcR11511)